crates/syntax/src/ast/edit.rs

```diff
@@ -459,6 +459,72 @@ impl ast::MatchArmList {
     }
 }
 
+impl ast::GenericParamList {
+    #[must_use]
+    pub fn append_params(&self, params: impl IntoIterator<Item = ast::GenericParam>) -> Self {
+        let mut res = self.clone();
+        params.into_iter().for_each(|it| res = res.append_param(it));
+        res
+    }
+
+    #[must_use]
+    pub fn append_param(&self, item: ast::GenericParam) -> Self {
+        let is_multiline = self.syntax().text().contains_char('\n');
+        let ws;
+        let space = if is_multiline {
+            ws = tokens::WsBuilder::new(&format!(
+                "\n{}    ",
+                leading_indent(self.syntax()).unwrap_or_default()
+            ));
+            ws.ws()
+        } else {
+            tokens::single_space()
+        };
```

Comment:
Let's just assume that `GenericParam`s are always on a single line (which I think is true most of the times). We'll fix that properly when we add real code formatter.